### PR TITLE
BREAKING!: change from demo to board

### PR DESCRIPTION
**Pull request by PeterShershov**
- Change all occurrences of `simulation`/`demo` in internal/public APIs and comments to `board`.
- Remove the `props` property that is no longer relevant in the new format.

### CODE REVIEW

File: packages/board-core/src/types.ts [hunk 1]
```diff
@@ -135,40 +135,24 @@ export interface IRenderableMetadataBase<HOOKS extends HookMap = HookMap>
      * sets the stage for the renderer.
      * this function has many side effects ( such as effecting window styles and sizes )
      *
-     * @returns canvas an html element for rendering into, cleanup a method for cleaing up the sideeffects
+     * @returns canvas an html element for rendering into, cleanup a method for cleaning up the side-effects
      *
      */
-    setupStage: (parentElement?: HTMLElement) => ReturnType<SetupSimulationStage>;
+    setupStage: (parentElement?: HTMLElement) => ReturnType<SetupBoardStage>;
```
Comment (AviVahl) [inline packages/board-core/src/types.ts:141]: perhaps rename `SetupBoardStage` to be more consistent with `BoardSetupFunction`?
maybe `BoardSetupStageFunction`? or shorten the postfix in both to `Fn`?

**Review by AviVahl — APPROVED** [review]
Looking good. consider the rename.

**Comment by AviVahl** [comment]
ah, file name is `simulation-fixtures.tsx` rename that too?

File: packages/react-board/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+```tsx
+// hello.board.tsx
+
+import { createBoard } from '@wixc3/react-board';
+import { Hello } from './hello';
+
+createBoard({
+  name: 'basic board',
+  componentType: Hello,
```
Comment (AviVahl) [inline packages/react-board/README.md:32]: rename key

File: packages/react-board/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  props: { name: 'World' },
```
Comment (AviVahl) [inline packages/react-board/README.md:33]: remove key

File: packages/react-board/README.md (hunk 2 above)
Comment (AviVahl) [reply packages/react-board/README.md:32]: and change value to render Hello passing name="World" or something

File: packages/board-plugins/src/react-context-plugin.tsx [hunk 4]
```diff
@@ -1,13 +1,13 @@
-import { createPlugin } from '@wixc3/simulation-core';
-import type { IReactDemo } from '@wixc3/react-simulation/src';
+import { createPlugin } from '@wixc3/board-core';
+import type { IReactBoard } from '@wixc3/react-board/src';
```
Comment (AviVahl) [inline packages/board-plugins/src/react-context-plugin.tsx:2]: not from /src

File: packages/react-board/test/fixtures/checkbox.board.tsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import React, { useState } from 'react';
+import { createBoard } from '../../src';
```
Comment (AviVahl) [inline packages/react-board/test/fixtures/checkbox.board.tsx:2]: use package name

File: packages/react-board/test/fixtures/simulation-fixtures.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { createBoard } from '@wixc3/react-board';
+
+export const MockComponent: React.FC<{ text: string }> = (props) => <div>{props.text}</div>;
+
+export const propsOnlyBoard = createBoard({
+    Board: () => <MockComponent text={'this is a test'} />,
```
Comment (AviVahl) [inline packages/react-board/test/fixtures/simulation-fixtures.tsx:7]: text="this is..."

File: packages/react-board/test/fixtures/simulation-fixtures.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    name: 'mock board',
+});
+
+export const overrideProps = {
+    text: 'override text',
+};
+
+export const boardWithWrapper = createBoard({
+    name: 'mock board',
+    Board: () => <MockComponent text={'mock board'} />,
```
Comment (AviVahl) [inline packages/react-board/test/fixtures/simulation-fixtures.tsx:17]: text="

File: packages/react-board/test/fixtures/simulation-fixtures.tsx [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+});
+
+export const boardWithEnvironmentProps = createBoard({
+    Board: () => <MockComponent text={'this is a test'} />,
```
Comment (AviVahl) [inline packages/react-board/test/fixtures/simulation-fixtures.tsx:21]: text="

File: packages/wcs-core/test/render-helpers.spec.ts [hunk 9]
```diff
@@ -1,102 +0,0 @@
...
-
-        it('returns correct JSX for a simulation with a wrapper', () => {
-            const renderedComponent = simulationToJsx(simulationWithWrapper);
-
-            expect(ReactTestUtils.isElement(renderedComponent)).to.equal(true);
-            expect(renderedComponent.type).to.eql(simulationWithWrapper.wrapper);
-        });
-    });
-
-    describe('setupSimulationStage', () => {
```
Comment (eladdo92) [inline packages/wcs-core/test/render-helpers.spec.ts:37]: are these tests still relevant?

Comment (PeterShershov) [reply packages/wcs-core/test/render-helpers.spec.ts:37]: nope. setupStage is tested in `packages/react-board/test/setup-stage.spec.tsx`

**Review by eladdo92 — APPROVED** [review]

File: packages/board-plugins/README.md [hunk 10]
```diff
@@ -31,12 +30,12 @@ createSimulation({
...
 ## tagsPlugin
 
-Adds relevant tags to the simulation. Useful for indexing the simulations.
+Adds relevant tags to the board. Useful for indexing the boards.
 
 ## cssVarsPlugin
 
-Accepts a record containing css var names and values to give them during the simulation.
-Adds those css variables to the convas style.
+Accepts a record containing css var names and values to give them during the board.
```
Comment (Eg0ld) [inline packages/board-plugins/README.md:37]: Should this be during rendering? During the board doesn't make sense.

**Pull request merged**